tests/system_metrics/test_system_metrics_logging.py

```diff
@@ -0,0 +1,76 @@
+import threading
+import time
+
+import mlflow
+from mlflow.system_metrics.system_metrics_monitor import SystemMetricsMonitor
+
+
+def test_manual_system_metrics_monitor():
+    with mlflow.start_run(log_system_metrics=False) as run:
+        system_monitor = SystemMetricsMonitor(
+            run.info.run_id,
+            sampling_interval=0.1,
+            samples_before_logging=5,
+        )
+        system_monitor.start()
+        thread_names = [thread.name for thread in threading.enumerate()]
+        # Check the system metrics monitoring thread has been started.
+        assert "SystemMetricsMonitor" in thread_names
+
+        time.sleep(2)
+    mlflow.end_run()
+    # Pause for a bit to allow the system metrics monitoring to exit.
+    time.sleep(1)
+    thread_names = [thread.name for thread in threading.enumerate()]
+    # Check the system metrics monitoring thread has exited.
+    assert "SystemMetricsMonitor" not in thread_names
+
+    mlflow_run = mlflow.get_run(run.info.run_id)
+    metrics = mlflow_run.data.metrics
+
+    expected_metrics_name = [
+        "cpu_utilization_percentage",
+        "system_memory_usage_megabytes",
+        "disk_usage_percentage",
+        "disk_usage_megabytes",
+        "disk_available_megabytes",
+        "network_receive_megabytes",
+        "network_transmit_megabytes",
+    ]
+    for name in expected_metrics_name:
+        assert name in metrics
+
+
+def test_automatic_system_metrics_monitor():
+    mlflow.enable_system_metrics_logging()
+    mlflow.set_system_metrics_sampling_interval(0.2)
+    mlflow.set_system_metrics_samples_before_logging(5)
```

Comment:
These lines mutate `os.environ` and might affect other tests. Can we use `monkeypatch` (https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.delenv) instead?

Reply:
fixed